src/OpenApi/src/Extensions/JsonTypeInfoExtensions.cs

```diff
@@ -0,0 +1,121 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO.Pipelines;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Serialization.Metadata;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.OpenApi;
+
+internal static class JsonTypeInfoExtensions
+{
+    private static readonly Dictionary<Type, string> _simpleTypeToName = new()
```

Comment:
Are there any other primitive-y types that should be added? For example I added `DateOnly`, `TimeOnly`, `(U)Int128` and some others to the latest release of Swashbuckle (admittedly for a different purpose): [JSON serializer context](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/blob/a97c53fd4dfe1a2ca2e2607959cf6c002b02c2e0/src/Swashbuckle.AspNetCore.SwaggerUI/SwaggerUIJsonSerializerContext.cs#L12-L41)

Reply:
`DateOnly`, `TimeOnly`, and a few others on the referenced link get handled by the JsonSchemaMapper for us. I'll do a few more comparisons but I think the set we have here is good enough.